pkg/skaffold/errors/errors.go

```diff
@@ -107,12 +108,18 @@ func IsOldImageManifestProblem(err error) (string, bool) {
 }
 
 func getErrorCodeFromError(phase Phase, err error) (proto.StatusCode, []*proto.Suggestion) {
-	if t, ok := err.(Error); ok {
+	uErr := func(err error) error {
+		if e := errors.Unwrap(err); e != nil {
+			return e
+		}
+		return err
+	}(err)
```

Comment:
This seems a bit complicated.  But I think this should use `errors.As()` which will try to unwrap recursively:
```
var t Error
if errors.As(err, &t) {
  return t.StatusCode(), t.Suggestions()
}
```

It seems a bit fraught to compare the regexps to the unwrapped error?

Reply:
Thanks. Done. I was looking for this function instead of `Unwrap`.